driver-core/src/main/java/com/datastax/driver/core/UserType.java

```diff
@@ -167,6 +168,16 @@ public DataType getFieldType(String name) {
     }
 
     @Override
+    public boolean isFrozen() {
+        return true;
+    }
+
+    @Override
+    boolean canBeDeserializedAs(TypeToken typeToken) {
+        return typeToken.isAssignableFrom(getName().javaType);
+    }
+
+    @Override
     public final int hashCode() {
         return Arrays.hashCode(new Object[]{ name, keyspace, typeName, byIdx });
     }
```

Comment:
In addition to these changes, the toString() method should be updated to surround the UDT with "frozen<>".